### PR TITLE
Syphilis - Doctor declaration processing

### LINKED ISSUE #14239: Syphilis - Lab messages processing / mapping
### Problem Description

As part of the Syphilis epic, a new disease (SYPHILIS, including congenital syphilis) is being introduced in SORMAS. Incoming laboratory messages (code questions SYPH-1 through SYPH-7) need to be mapped correctly into SORMAS case classification, symptom/stage/infectiousness fields, and test results. The mappings are defined in the "The metadata information is mapped as follows" section of the SORMAS requirements document for Syphilis.

### Proposed Solution

Configure code question mappings so that each SYPH-1 through SYPH-7 code question maps to the appropriate SORMAS field or sample/test, based on the coded value lists received from the laboratory (LNS). SYPH-1 to SYPH-3 update case-level fields (classification, stage, infectiousness) rather than creating tests; SYPH-4 to SYPH-7 create/update tests on a clinical sample (or serum for SYPH-7) and set case classification to confirmed.

#### Code question mappings

- [x] **SYPH-1 (Laboratory Result)** — variable `LaboratoryResult`, coded value list `LabResults` — not a laboratory test result, maps to case classification:
  - [x] Coded value `CONF` (Confirmed) → Case classification = **Confirmed**
  - [x] Coded value `PROB` (Probable) → Case classification = **Probable**
  - [x] Coded value `NA` (Not Applicable) → Case classification = **Not yet classified**
  - [x] Coded value `UNK` (Unknown) → Case classification = **Not yet classified**

- [x] **SYPH-2 (Stage – simple / Infectiousness)** — variable `StageSYPH`, coded value list `StageSYPHsimple` — not a laboratory test result, maps to the Infectiousness field (Symptoms 3.3.E):
  - [x] Coded value `I` (Infectious) → Infectiousness = **Infectious**
  - [x] Coded value `NI` (Not Infectious) → Infectiousness = **Not Infectious**
  - [x] Coded value `UNK` (Unknown) → Infectiousness = **Unknown**

- [x] **SYPH-3 (Stage – detailed)** — variable `StageSYPHdetailed`, coded value list `StageSYPHdetailed` — not a laboratory test result, maps to the Stage field (Symptoms 3.3.E):
  - [x] Coded value `P` → Stage = **Primary syphilis**
  - [x] Coded value `S` → Stage = **Secondary syphilis**
  - [x] Coded value `EL` → Stage = **Early latent syphilis**
  - [x] Coded value `L` → Stage = **Late latent syphilis** (because unknown is assumed to be Late latent)
  - [x] Coded value `LL` → Stage = **Late latent syphilis**
  - [x] Coded value `UNK` → Stage = **Unknown**
  - Note: SORMAS Stage dropdown also includes **Tertiary syphilis** (`T`), not sent as a coded value by the lab but usable manually

- [x] **SYPH-4 (Dark-field microscopy)** is configured to map when result = positive:
  - [x] Sample = **Clinical sample**
  - [x] Test = **Dark-Field Microscopy**
  - [x] Test category = **Microscopy & Staining**
  - [x] Test result = **Positive**
  - [x] Case classification set to **Confirmed**

- [x] **SYPH-5 (Direct immunofluorescence)** is configured to map when result = positive:
  - [x] Sample = **Clinical sample**
  - [x] Test = **Direct fluorescent antibody** 
  - [x] Test category = **Microscopy & Staining**. 
  - [x] Test result = **Positive**
  - [x] Case classification set to **Confirmed**

- [x] **SYPH-6 (NAAT)** is configured to map when result = positive:
  - [x] Sample = **Clinical sample**
  - [x] Test = **NAAT**
  - [x] Test result = **Positive**
  - [x] Case classification set to **Confirmed**

- [x] **SYPH-7 (Screening antibody test + IgM/non-TP confirmation)** is configured to map when result = positive:
  - [x] Sample = **Serum**
  - [x] Two linked tests are created under Test category = **Serological test**:
    - [x] Subcategory **Non-treponemal tests (NTT)** — Result = **Positive**
    - [x] Subcategory **Treponemal tests (TT)** — Result = **Positive**
  - [x] Laboratory Method left blank for both created tests
  - [x] Case classification set to **Confirmed**

#### Edge case handling

- [x] SYPH-1 values `NA` and `UNK` both resolve to case classification **Not yet classified** — do not leave classification blank
- [x] SYPH-2/SYPH-3 must not override a stage/infectiousness value already set manually if a conflicting rule applies (align with existing symptom-driven prefill logic in 3.3.E)
- [x] SYPH-4 through SYPH-7 always force case classification to **Confirmed**, regardless of any prior classification value
- [x] Code question messages from LNS should support simultaneous multi-value reporting (e.g., SYPH-2 + SYPH-3 + SYPH-6 in one message)

### Additional Information

- SYPH-1 through SYPH-3 are **not laboratory test results** — they carry case-level metadata (classification, stage, infectiousness) and must not create Test entities.
- SYPH-4 through SYPH-7 **do** create/update laboratory tests and always set the case classification to confirmed.

### LINKED ISSUE #14240: Syphilis - Doctor declaration processing
# Doctor's notification mapping

## Problem description

Problem Description
As part of the Syphilis epic, a new disease (SYPHILIS, including congenital syphilis) is being introduced in SORMAS. Incoming Doctor notifications need to be mapped correctly into SORMAS. The mappings are defined in the SORMAS requirements document for Syphilis.

## XML field mapping

Sample doctor's notification payload:

```xml
<mCAlopecieDiffuse>true</mCAlopecieDiffuse>
<mCChancres>true</mCChancres>
<mCCondylomesVeneriens>true</mCCondylomesVeneriens>
<mCEnantheme>true</mCEnantheme>
<mCEruptionMaculopapuleuse>true</mCEruptionMaculopapuleuse>
<mCLymphadenopathieGeneralisee>true</mCLymphadenopathieGeneralisee>
<mCManifestationsNeurologiques>true</mCManifestationsNeurologiques>
<mCManifestationsOculaires>true</mCManifestationsOculaires>
<mCAutre>Other_sympt</mCAutre>
<siteInfGenital>true</siteInfGenital>
<siteInfAnoRectal>true</siteInfAnoRectal>
<siteInfPharyngeal>true</siteInfPharyngeal>
<siteInfAutre>Other_siteInfection</siteInfAutre>
<stageDeLinfection>infectieux</stageDeLinfection>
<transmissionProbable>contactHeterosexuel</transmissionProbable>
<contactTravailleurDeSexe>inconnu</contactTravailleurDeSexe>
<statutVih>positif</statutVih>
<paysDeNaissance>LU</paysDeNaissance>
<nationalite>LU</nationalite>
```

### Case Data

- [x] `statutVih` (value `positif`) <-> `HealthConditions.hivStatus` = Positive

### Symptoms

- [x] `mCAlopecieDiffuse` <-> `Symptoms.patchyAlopecia`
- [x] `mCChancres` <-> `Symptoms.chancre`
- [x] `mCCondylomesVeneriens` <-> `Symptoms.condylomaVenereum`
- [x] `mCEnantheme` <-> `Symptoms.mucousPatches`
- [x] `mCEruptionMaculopapuleuse` <-> `Symptoms.maculopapularRash`
- [x] `mCLymphadenopathieGeneralisee` <-> `Symptoms.generalizedLymphadenopathy`
- [x] `mCManifestationsNeurologiques` <-> `Symptoms.neurologicalManifestations`
- [x] `mCManifestationsOculaires` <-> `Symptoms.ocularManifestations`
- [x] `mCAutre` (boolean flag) <-> `Symptoms.otherClinicalPresentation`
- [x] `mCAutre` (text value, e.g. `Other_sympt`) <-> `Symptoms.otherClinicalPresentationText`
- [x] `siteInfGenital` <-> `SyphilisInfectionSite.GENITAL`
- [x] `siteInfAnoRectal` <-> `SyphilisInfectionSite.ANORECTAL`
- [x] `siteInfPharyngeal` <-> `SyphilisInfectionSite.PHARYNGEAL`
- [x] `siteInfAutre` (text value, e.g. `Other_siteInfection`) <-> `SyphilisInfectionSite.OTHER`
- [x] `stageDeLinfection` (value `infectieux`/ `inconnu`) <-> `Symptoms.syphilisInfectiousness`
- [x] `StadeDetailleInfection` <-> `Symptoms.syphilisStage`

### EpiData

- [x] `transmissionProbable` <-> `EpidemiologicalData.probableRouteOfTransmission`
- [x] `contactTravailleurDeSexe` <-> `EpidemiologicalData.contactWithSexWorker`

### EpiData

Those 2 could be true for congenital syphilis, but otherwise cannot be used ? 
- [ ] `paysDeNaissance` <-> `EpidemiologicalData.motherCountryOfBirth`
- [ ] `nationalite` <-> `EpidemiologicalData.motherCitizenship`

### PR DESCRIPTION
**Pull request by Pa-Touche**
Not sure at all of those changes. Those are required to make the adapter build. 
Pathogen tests will be done by someone else

Fixes #14240

## Summary by CodeRabbit

* **New Features**
  * Expanded external-message support for syphilis, HIV status, STI prophylaxis, transmission route, sex-worker contact, and maternal birth information.
  * Person and maternal citizenship and country-of-birth details can now be captured and transferred.
  * Added reporting filters for cases included or excluded from reporting.
* **Improvements**
  * Syphilis-related information is now reflected in case and epidemiological data.
  * Jaundice is recognized as a syphilis complication.
  * Added labels for newly supported health and demographic fields.

### CODE REVIEW

File: sormas-api/src/main/java/de/symeda/sormas/api/externalmessage/ExternalMessageDto.java [hunk 1]
```diff
@@ -301,6 +321,12 @@ public class ExternalMessageDto extends SormasToSormasShareableDto {
 	private String clusterTypeText;
 	private Boolean clusterRelated;
 
+	private HivStatus hivStatus;
+	private ProbableRouteOfTransmission probableRouteOfTransmission;
+	private YesNoUnknown contactWithSexWorker;
+	private CountryReferenceDto motherCountryOfBirth;
+	private CountryReferenceDto motherCitizenship;
```
[the remark this answers is not in the document]
Comment (Pa-Touche) [reply sormas-api/src/main/java/de/symeda/sormas/api/externalmessage/ExternalMessageDto.java:328]: required

File: sormas-backend/src/test/java/de/symeda/sormas/backend/externalmessage/ExternalMessageFacadeEjbMappingTest.java [hunk 2]
```diff
@@ -150,6 +150,9 @@ public void testFromDto() {
 		when(countryService.getByReferenceDto(source.getPersonCountry())).thenReturn(country);
 		when(facilityService.getByReferenceDto(source.getPersonFacility())).thenReturn(facility);
 
+		when(countryService.getByReferenceDto(source.getPersonBirthCountry())).thenReturn(country);
+		when(countryService.getByReferenceDto(source.getPersonCitizenship())).thenReturn(country);
```
[the remark this answers is not in the document]
Comment (Pa-Touche) [reply sormas-backend/src/test/java/de/symeda/sormas/backend/externalmessage/ExternalMessageFacadeEjbMappingTest.java:156]: indeed values are now set

**Review by roldy — APPROVED** [review]
